### PR TITLE
feat(report): add report feature

### PR DESCRIPTION
**Pull request by rajaSahil**
**Purpose of PR?**:
- Add support for nested struct output for devsecops.

**Does this PR introduce a breaking change?**
- None

**If the changes in this PR are manually verified, list down the scenarios covered:**:
- Have not tested it locally due to nil pointer exception in codebase

**Additional information for reviewer?** :
- Please ignore all the code repetition. Have pushed the changes due to highest priority item and will be doing refactoring as a part of different PR.
- Have also not considered enhancements, there can be many enhancements possible in this codebase.
- Please ignore commented code in PR.

**Checklist:**
- [ ] Bug fix. Fixes #<issue number>
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [x] PR Title follows the convention of  `<type>(<scope>): <subject>`
- [ ] Commit has unit tests
- [ ] Commit has integration tests

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -14,4 +14,4 @@ kubearmor_policies*.yaml
 cilium_policies*.yaml
 knox_net_policies*.yaml
 src/accuknox*.db*
-
+src/src
```
Comment (nyrahul) [inline .gitignore:17]: why this?

File: src/conf/local-file.yaml [hunk 2]
```diff
@@ -34,7 +34,7 @@ application:
 
 observability: 
   enable: true
-  cron-job-time-interval: "0h0m10s"         # format: XhYmZs
+  cron-job-time-interval: "0h0m1s"         # format: XhYmZs
```
Comment (nyrahul) [inline src/conf/local-file.yaml]: Please revert.

File: src/conf/local-file.yaml [hunk 3]
```diff
@@ -54,7 +54,7 @@ purge-old-db-entries:
 database:
   driver: sqlite3
   host: 127.0.0.1
-  port: 3306
+  port: 3308
```
Comment (nyrahul) [inline src/conf/local-file.yaml]: Please revert

File: src/protobuf/v1/report/report.proto [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+  string ResourceType=1;
+  string ResourceName=2;
+  MetaData MData=3;
+  SummaryData SumData=4;
+}
+
+message SummaryData{
+
+  repeated observability.SysProcFileSummaryData ProcessData = 1;
+  repeated observability.SysProcFileSummaryData FileData = 2;
```
Comment (nyrahul) [inline src/protobuf/v1/report/report.proto:54]: We need SyscallData as well as part of this.

**Review by nyrahul — CHANGES_REQUESTED** [review]

File: src/conf/local-file.yaml (hunk 2 above)
Comment (seswarrajan) [inline src/conf/local-file.yaml]: Why was this change from 10s to 1s ?

File: src/observability/kubearmor.go [hunk 5]
```diff
@@ -212,14 +212,14 @@ func GetKubearmorSummaryData(req *opb.Request) ([]types.SysObsProcFileData, []ty
...
-	sysSummary, err := libs.GetSystemSummary(CfgDB, types.SystemSummary{
+	sysSummary, err := libs.GetSystemSummary(CfgDB, &types.SystemSummary{
 		PodName:       req.PodName,
 		NamespaceName: req.NameSpace,
 		ContainerName: req.ContainerName,
 		ClusterName:   req.ClusterName,
 		Labels:        req.Label,
 		Deployment:    req.DeployName,
-	})
+	}, nil)
```
Comment (seswarrajan) [inline src/observability/kubearmor.go:222]: Why is this hardcoded to nil ?

**Review by seswarrajan — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (rajaSahil) [reply .gitignore:17]: When we run go build command to create binary, binary name is src. That's why I have added it over here.

File: src/conf/local-file.yaml (hunk 2 above)
Comment (rajaSahil) [reply src/conf/local-file.yaml]: This was done for testing in local.

File: src/observability/kubearmor.go (hunk 5 above)
Comment (rajaSahil) [reply src/observability/kubearmor.go:222]: Since we don't need report options for summary.

File: src/protobuf/v1/report/report.proto (hunk 4 above)
Comment (rajaSahil) [reply src/protobuf/v1/report/report.proto:54]: Will add it in the subsequent PRs.

**Review by nyrahul — APPROVED** [review]

**Pull request merged**